### ISSUE TITLE
As Game Designer I wish to control how time is presented

### ISSUE THREAD

**Issue by IanMayo** [serge-web/serge#954]
Serge includes some time data in the top-right of the Player page.

This data is configured from the Overview tab of the Admin pages for the wargame. Here is the current layout:
![image](https://user-images.githubusercontent.com/1108513/126675641-a4762fcd-9b95-46ee-8935-a88294685e6e.png)

Game Designers would like to have greater control over the display of time, including these configuration parameters:
* Game turn length (as described [here](https://github.com/serge-web/serge/issues/991))
* Game-start date & time (once the game is playing, this will actually allow the Game Control to modify _current_ game time).
* Initial game phase:`Planning` or `Adjudication`.  (again, this will allow Game Control to modify current phase one game playing)
* How to format game time (drop-down list of date, date-time, and time formats - see below)
* Whether to show planning time remaining (we don't need format  for this, hours/mins is the only format required)
* How to display game time:
    * time of start of turn
    * time of end of turn
    * start and end times for turn
    * do not display game time (for some games there is just turn counters)
    
The above controls will clutter the Overview page.  We should probably re-plan the controls on the Overview page to keep them all logical & tidy.  The use of `fieldset` boxes may help organise the data.

The "Game Overview" box has not been used by any Game Designers yet - it would be ok to reduce the space it occupies.

## Date formats
* yyyy/mmm/dd
* yy/mm/dd
* yy/mm/dd hh:mm
* hh:mm
* ddhhmm (this is a special format used in the Navy)

**Comment by IanMayo** [comment]
Hello @ShendMikullovci - please notice I've added *game start date-time* to the above list of required data controls.

**Comment by IanMayo** [comment]
Hello @ShendMikullovci , it would be useful to handle #955 in this issue.

**Comment by ShendMikullovci** [comment]
Added controls and details:
![Game Overview Optimization](https://user-images.githubusercontent.com/85491794/129016197-b6f474cd-0530-48a5-a4f0-7824dcb0637a.png)

Much more complete display with an abundance of options available. New “Formatting” fieldset added to allow the choice of turn and date/time format. 
Time Settings elaborated as discussed. I’ve also added a tick option to the “Time Warning” feature in case the game designer decides to remove the option from the game.
Due to the familiarity of the Save button being on the lower end of screens, I’ve decided to lower the “Save Overview” button to the lower right side of the display: After running through the options and details, you naturally end on the bottom of the screen where the button is highlighted and draws attention.
As per Issue #955, I’ve added the required controls (i.e., Turn number & Turn Presentation) onto the overview page.

Let me know what you guys think!

**Comment by IanMayo** [comment]
Thanks for the design. Feedback:
1. `Phase Time Remaining` is duplicated
2. We need to be able to switch off the concept of `Planning time`, if their allowed time is being managed outside Serge. This may be handled through a tick-box next to `Allowance`.
3. I think it would be more economical use of screen space for the `Formatting` blocks to be drop-downs. Then we only see the available options when we're _engaged_ with that attribute
4. We don't control turn start and end time. We control turn start and turn length (which effectively gives us the same results)
5. `Planning` and `Adjudication` _could_ be represented as either a pair of radio buttons or a drop-down list
6. I think the time display strategy needs supporting (image below)


![image](https://user-images.githubusercontent.com/1108513/129018873-260d7f25-d230-4816-8cbb-743965566ac4.png)

**Comment by ShendMikullovci** [comment]
After review of feedback and discussion with Ian, I’ve made the following adjustments:
(Bullet points correspond to Ian's feedback points).

1. “Phase Time Remaining” item has been removed completely due to redundancy (allowance represents the same thing in-game).

2. Tickbox on “Allowance” to make it optional.

3. Due to varying needs of the clients, we decided to restructure the design into three drop-down windows containing the following info:
a. “Time Settings”: Depending on whether the client needs/wants to display time, they have the option to enable or disable the feature. 
b. “Turn Settings”: Contains time and phase elements.
c. “Formatting”: Settings box for the game designer to select which time format they prefer. 

If the “Display Time” feature is enabled, all time-related information is underlined, clickable and modifiable:

![Game Overview Optimization With Displayed Time](https://user-images.githubusercontent.com/85491794/129071073-cb8a64a9-a31e-4f06-918b-9e301f3302cd.jpg)

If the feature is disabled, all time-related information is dimmed and unclickable:

![Game Overview Optimization Without Displayed Time (1)](https://user-images.githubusercontent.com/85491794/129071155-d89e3d6b-a21e-4264-9f11-f5502405df71.jpg)

“Turn Number” remains due to the fact that it’s not directly world-time-related information and could still be required even if there’s no need to display time.

4. “Turn Duration” replaces “Turn End Time” as a more representative terminology.

5. Radio buttons added to the “Game Phase” feature to reinforce the dissection between phases in the game.

6. Restructured design frame to support the needs of the client. In the “Formatting” window on the right hand side, the game designer is able to select among the different “Turn Format” and “Date/Time Format” displays to match their needs. For the upcoming wargame, the formats “Pairs” and “YYYY/MM/DD” could be suitable.

7. I also thought of making each of the upper rectangles minimizable in order to avoid clutter: If the game designer wishes to return to the overview page to consult the “Game Description and Objectives” window, they’re able to minimize the windows above to avoid distraction. Illustration below:

![Game Overview Optimization With Minimized Windows](https://user-images.githubusercontent.com/85491794/129071390-075fe774-9203-4b82-b38d-6bbe283113b3.png)

Let me know what you think!

**Comment by IanMayo** [comment]
Thanks for the update @ShendMikullovci 

Couple of items:
1. We're not allowing the Game Designer to choose to display the `Turn end time` or `Turn start and end times`. I _think_ this could be supported by changing `Display time` radio buttons to a ` Game time` drop-down with the 4 options listed [above](https://github.com/serge-web/serge/issues/954#issuecomment-896737096)
2. IMHO `Turn start time` and `Turn duration` belong in the `Time Settings` block, since they related to the display of time. Note: they will get disabled if `Display time` is `off`.
3. There may be a different logical organisation.  We _may_ be able to put the contents of `Formatting` back near the values they related to.  `Turn Format` could logically go beneath `Turn number`, and `Date/time format` could logically go beneath `Display time` (and be disabled if `Display time` is `Off`).
4. I'm not a fan of the collapsed sections - since the Game Designer will have to manually expand each one to review the settings. I think we have enough screen space to allow them to all be displayed.

**Comment by ShendMikullovci** [comment]
Noted! I edited the design and this is how it looks:
![Game Overview Optimization Edited (1)](https://user-images.githubusercontent.com/85491794/129185714-3272633f-8019-4911-a092-55cd3279ed9e.png)

1. Changed “Display Time” to “Game Time” with the 4 aforementioned options listed. It does take quite some room in the display, let me know your thoughts.
2. “Turn Start Time” and “Turn Duration” added to the “Time Settings” window. Dimmed here to illustrate the “Do Not Display Time” feature. 
Question: Should we disable “Allowance” and “Warning” when “Do Not Display Time” is enabled? 
3. “Turn Format” added under “Turn Number” in “Turn Settings”. “Date/Time Format” clutters the “Time Settings” window too much in my opinion.
4. Removed the collapsing feature.

Let me know what you think!

**Comment by IanMayo** [comment]
Thanks @ShendMikullovci  :-)

The `Date/Time Format` control could be a drop-down list instead of a radio-button list, and take up less space. In this way, the  Game Designer only sees the _available_ formats when the drop-down is expanded - but they remain able to view the _current_ format all the time.

The `Game time` control could also switch to drop-down, and consume less space.

No, it's not easy to draw `Drop-downs` in drawing tool.  One way is to show them in a white rectangle that sits above the other controls that would be hidden.

With the above two changes, it seems achievable to have the `Date/time format` in the block that is currently greyed out below `Game Time`.

**Allowance** and **Warning**.  No, these don't need to be disabled when `Do not display time` is enabled.

Hope this all helps :-)

**Comment by ShendMikullovci** [comment]
Made the adjustments! 
1. Here’s how the design looks with the drop-downs closed:
![Game Overview Optimization Drop Down List Closed](https://user-images.githubusercontent.com/85491794/129233983-4e56c73d-da7b-4b21-ad3f-30216d76d1b6.png)

2. Here’s how the “Game Time” drop-down looks like opened: 
![Game Overview Optimization _Game Time_ Drop Down List](https://user-images.githubusercontent.com/85491794/129234052-e1cb54e4-5ab1-4e86-b91a-1930878c2791.png)

Underlined sentences represent clickable and/or modifiable areas. Highlighted in grey is the display mode in design. I have to admit I was hesitant about whether I should add tick boxes/radio buttons for the display modes or not. What are your thoughts?

3. Here’s how the “Date/Time Format” drop-down looks like opened:
![Game Overview Optimization _Date_Time Format_ Drop Down List](https://user-images.githubusercontent.com/85491794/129234201-c812d19e-c13c-4d43-bc31-26c2f4a519ea.png)

I kept the tick boxes here for comparison with the “Game Time” drop-down. We can select whatever you think is more useful and aesthetically pleasing.
Let me know your thoughts!

**Comment by IanMayo** [comment]
Sure, I understand the doubt about tick-marks.  Since only one item in each drop-down can be selected, I don't think we need them. There will just be one shaded item when the drop-down list is expanded.

One element that isn't quite clear, though it may be a consequence of the drawing app:  in the `Game Turn` drop-down, only the 4 game-turn style items will be in the drop-down menu.  In the second graphic the other items would sit on the page - not in the dropdown.

**Comment by ShendMikullovci** [comment]
Noted. Here’s how the design looks with the exterior elements:
![Game Overview Optimization Drop Down List Closed Edited](https://user-images.githubusercontent.com/85491794/129444785-198bb53d-cfcb-4614-92b4-c64501d44a74.png)

Here’s how it looks once “Game Time” drop-down is open (considering “Turn Start/End Times” is selected):
![Game Overview Optimization _Game Time_ Drop Down List Edited](https://user-images.githubusercontent.com/85491794/129444796-4c0e9681-673e-4aeb-8ab7-b6bb0da39c83.png)

With “Do Not Display Time” selected we would have the following look:
![Game Overview Optimization _Game Time_ Drop Down List Edited with _Do Not Display_ Selected](https://user-images.githubusercontent.com/85491794/129444805-f0a984f7-01ee-4749-93bc-928a03aaf2be.png)

Let me know what you think!

**Comment by IanMayo** [comment]
Thanks for that @ShendMikullovci  :-)

Yes - it's a nice compact (but logical) layout.

One (hopefully last :-) ) suggestion.

It's about the two drop-down lists in `Time settings`.

The current strategy displays the label (Game Time) when the list is collapsed.  This obscures value `Time of start of turn`.

It would be better to display `Game time` as a text-label to the left of the drop-down, and the current value (e.g. `Do not display time`) as the value of the drop-down when collapsed.  This allows the Game Designer to review the settings without having to expand the list.  

I see we do have enough horizontal space to allow this without any negative impact.

**Comment by ShendMikullovci** [comment]
When you say to the left of the drop-down, do you mean outside of the entire "Time Settings" window?

**Comment by IanMayo** [comment]
Hello :-)

![image](https://user-images.githubusercontent.com/1108513/130058558-695c8c8f-4276-4178-bf81-ed4aeb6d5804.png)

In the above shot you can see the label (`Book category`) is shown to the left of the drop-down, and the current value `Computers and Internet` is shown in the drop-down.

The combo box is open in that screenshot, but if we imagine it collapsed, we will be able to see both the label/name and the current value.

**Comment by ShendMikullovci** [comment]
Great idea! I tried to implement it. This is how it looks:
![Game Overview Optimization Text Labels Drop Down List ](https://user-images.githubusercontent.com/85491794/130319119-d5ebe234-a298-4dea-9ddf-20ce78cd4c7c.png)
This is the default display once “Turn Start/End Times” and “DDHHMM” are selected for Game Time and Date/Time Format. The user can therefore instantly see the currently selected options without having to drop the list down.

![Game Overview Optimization Text Labels Game Time Drop Down List Open](https://user-images.githubusercontent.com/85491794/130319134-6ead5eca-8379-4d1b-906b-03486c972e99.png)
Once the user does drop the list down, the selected option remains highlighted to maintain awareness of the selected option. 

![Overview Optimization Text Labels Drop Down List Date_Time Format](https://user-images.githubusercontent.com/85491794/130319149-67d2cfc9-9a65-4c2a-8330-39235250e5e4.png)
Same concept for the Date/Time Format. I decided to remove the tick boxes from the drop-down list due to clutter and incoherence with the Game Time list.

![Game Overview Optimization Text Labels Drop Down List _Do Not Display Time_](https://user-images.githubusercontent.com/85491794/130319165-e08e3a34-ec51-4129-a33c-cc4a3c1a34f1.png)
Once “Do Not Display Time” is selected, the “Game Time”, “Turn Start Time” and “Turn Duration” configurations are dimmed and unclickable.

I was thinking: In the “Do Not Display Time” scenario, do we disable the “Date/Time Format” option entirely as well? 

Let me know your thoughts!

**Comment by IanMayo** [comment]
Correct.  In "Do not display time" mode we should not display date/time format.

One other item.  I _don't think_ we need `Game time` and `Game start time`.  Aah, hold on. Yes we do :-) We need the _original_ start time in case they Game Control wants to restart from the original start time.

In that case, the _logical_ order is for `Game start time` to come before `Game time` - since once the game is playing the `Game time` will be _after_ the `Game start time`.

One last item on the UI.  You have created labels for the drop-down lists.  I believe these labels should sit _inside_ your white rounded rectangles.  Hmm, _maybe_ we don't need the white rounded rectangles any more - since it's just one set of controls.

**Comment by ShendMikullovci** [comment]
Good idea. The rectangles have been removed and it looks better. Furthermore, it differentiates the controls from the “Game Description and Objectives” field where text can be interacted with. I also dimmed the date/time format on the “Do Not Display Time” option. Finally, I decided to add “Start Time” as such to avoid confusion, given that we also have “Turn Start Time”. 
This is how it all looks:

![Game Overview Optimization Text Labels Drop Down List No White Rectangle ](https://user-images.githubusercontent.com/85491794/132187988-208dd98d-c0c1-41be-a930-c4bd77cfeb66.png)

![Game Overview Optimization Text Labels Drop Down List _Do Not Display Time_ No White Rectangle](https://user-images.githubusercontent.com/85491794/132188003-85b8409a-5170-4ff7-b496-8622060fee78.png)

![Game Overview Optimization Text Labels Game Time Drop Down List Open No White Rectangle](https://user-images.githubusercontent.com/85491794/132188026-9748b6fc-1809-4999-9182-7e7f5b84a3de.png)

Let me know what you guys think!

**Comment by IanMayo** [comment]
Thanks @ShendMikullovci 

Couple of items:
- I think it's worth putting the `Game Time` drop-down first, since if selection is `Do not display time` is selected then the `Start time` (as well as those in your screenshot) can be disabled
- `Start time` should actually be date _and_ time.  The current wargame is planned to start several years into the future.

**Comment by ShendMikullovci** [comment]
What type of format would you prefer for the "Start Time"? "YYYY/MM/DD - 00:00:00"?

**Comment by IanMayo** [comment]
Hello :-). 

I don't think games will run in terms of centuries, or in terms of seconds.  So I think we can shorten it to: `YY/MM/DD - 00:00`.

**Comment by ShendMikullovci** [comment]
Alright. Here’s how it looks:
![Game Overview Optimization Text Labels Game Time Drop Down List Open No White Rectangle (1)](https://user-images.githubusercontent.com/85491794/132993102-979b6e5a-2a5a-4fcc-9c44-5f2f0f1b8d0c.png)
![Game Overview Optimization Text Labels Drop Down List No White Rectangle ](https://user-images.githubusercontent.com/85491794/132993104-efee8c4a-5a67-4148-83f6-63cd6e61b1e3.png)
![Game Overview Optimization Text Labels Drop Down List _Do Not Display Time_ No White Rectangle](https://user-images.githubusercontent.com/85491794/132993105-e5d04409-a30c-44b7-9314-412fe6d987e5.png)

Let me know what you guys think!

**Comment by IanMayo** [comment]
Yes, that is great - thanks @ShendMikullovci

**Comment by ShendMikullovci** [comment]
This is how a potential solution to issue #991 looks in the “Game Overview” window:
![Game Overview Optimization 3 Tab Addition](https://user-images.githubusercontent.com/85491794/133891867-ae326803-7c67-4d11-88e9-6dc103bfb3f8.png)

Due to the need to differentiate two main types of time, “Time Settings” becomes “In-Game Time”. Elapsed time is not shown and “Turn Start Time” becomes “Current Time”. “Turn Duration” and “Format” enter the tab window for better efficiency and decluttering. 

“Allowance” and “Warning” get their own window titled “Real-World Time”. The “Turns Settings” window stays identical except the title, which becomes “Turns”.

Note: I decided to remove the “Game Description and Objectives” window after not having found anything in our notes that points to its significance. We can restore it anytime if needed.

To review the tabs in greater detail, refer to issue #991.

Let me know what you think!
